packages/providers/src/provider.ts

```diff
@@ -205,28 +207,75 @@ export type ProviderCallParams = {
   utxoValidation?: boolean;
 };
 
+/**
+ * URL - Consensus Params mapping.
+ */
+type ConsensusParamsCache = Record<string, ConsensusParameters>;
+
 /**
  * A provider for connecting to a node
  */
 export default class Provider {
   operations: ReturnType<typeof getOperationsSdk>;
   cache?: MemoryCache;
+  consensusParamsCache: ConsensusParamsCache = {};
 
   /**
    * Constructor to initialize a Provider.
    *
    * @param url - GraphQL endpoint of the Fuel node
+   * @param consensusParams - Consensus parameters of the Fuel node
    * @param options - Additional options for the provider
+   * @hidden
    */
   constructor(
     /** GraphQL endpoint of the Fuel node */
     public url: string,
+    consensusParams: ConsensusParameters,
     public options: ProviderOptions = {}
   ) {
     this.operations = this.createOperations(url, options);
+    this.consensusParamsCache[url] = consensusParams;
     this.cache = options.cacheUtxo ? new MemoryCache(options.cacheUtxo) : undefined;
   }
 
+  /**
+   * Creates a new instance of the Provider class. This is the recommended way to initialize a Provider.
+   * @param url - GraphQL endpoint of the Fuel node
+   * @param options - Additional options for the provider
+   */
+  static async connect(url: string, options: ProviderOptions = {}) {
+    const consensusParameters = await this.getConsensusParamsWithoutInstance(url);
+    const provider = new Provider(url, consensusParameters, options);
+    return provider;
+  }
+
+  /**
+   * Re-fetches the consensus parameters from the chain for the current URL.
+   */
+  async invalidateConsensusParamsCache() {
+    const { consensusParameters } = await this.getChain();
+    this.consensusParamsCache[this.url] = consensusParameters;
+  }
```

Comment:
would be nice to reuse the `connect` method

```suggestion
  async refetchConsensusParams() {
    // clear the cache for current URL
    delete this.consensusParamsCache[this.url];
    
    // connect will handle getting the consensusParams if it's not cached
    await this.connect();
  }
```